src/rust/generator-executor/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+use async_stream::stream;
+use futures_util::pin_mut;
+use generator_sdk::{
+    client::{
+        CacheBuilder,
+        GeneratorClient,
+        GeneratorClientError,
+        TokioAsyncResolver,
+    },
+    ClientConfig as GeneratorClientConfig,
+};
+use grapl_utils::future_ext::GraplFutureExt;
+use plugin_work_queue::client::PluginWorkQueueServiceClient;
+use rust_proto::plugin_work_queue::{
+    AcknowledgeGeneratorRequest,
+    ExecutionJob,
+    GetExecuteGeneratorRequest,
+    GetExecuteGeneratorResponse,
+};
+use structopt::StructOpt;
+use tokio::time::{
+    sleep,
+    Duration,
+};
+use tokio_stream::{
+    Stream,
+    StreamExt,
+};
+
+#[derive(StructOpt)]
+pub struct GeneratorExecutorConfig {
+    #[structopt(env)]
+    plugin_work_queue_address: String,
+    // Path to the certificate to be used when validating the generator plugin TLS sessions
+    #[structopt(env)]
+    generator_certificate_path: std::path::PathBuf,
+    #[structopt(flatten)]
+    generator_client_config: GeneratorClientConfig,
+}
+
+fn make_job_stream(
+    mut work_queue_client: PluginWorkQueueServiceClient<tonic::transport::Channel>,
+) -> impl Stream<Item = (ExecutionJob, i64)> {
+    stream! {
+        loop {
+            let message = match work_queue_client.get_execute_generator(GetExecuteGeneratorRequest {}).await {
+                Ok(message) => message,
+                Err(e) => {
+                    tracing::error!(message="Failed to retrieve generator job", error=?e);
+                    // check what kind of error it is and retry it accordingly
```

Comment:
Is this actually a `TODO`? If not, how is the checking and retrying actually achieved by this code?